### DISCUSSION TITLE
Routing with domains

### DISCUSSION THREAD

**Discussion by dgavey** [adonisjs/core#4118] [Help] [answered]
I'm not sure if this is a bug, or expected behaviour.  So asking here before I make a github issue. 

In my routes.ts I have a Route group with a domain like this

```
Route.group(() => {
  Route.get('/home', 'HomeController.index').as('app.home')
}).domain(`app.mydomain.com`)
```

Which works as expected.  However if I'm in an edge template and I use the `route` helper like so `{{route('app.home')}}`it throws an error and will not find my route.  I have to use  `{{route('app.home', null, {domain:'app.mydomain.com'})}}` instead.

This is very awkward, and I'm not sure if it's a bug or expected.   Thoughts?

**Comment by McSneaky** [comment, marked as answer]
Looks like expected

Adonis can't route, if it doesn't know your domain, coz you might want to direct from `foo.com/products` to `foo.com/cart` or `foo.com/products` to `bar.com/cart`, so you have to provide domain

That `null` in there is lil bit annoying, but it looks like expected thing

**Comment by dgavey** [comment]
Thanks for the answer.  Would be nice if by providing names for the routes that they could only be used once, then the route helper would be simpler to use in the templates.  Either way I have workarounds.
